cost/aws/reserved_instances/recommendations/aws_reserved_instance_recommendations.pt

```diff
@@ -283,20 +262,79 @@ datasource "ds_ri_recommendations" do
 end
 
 datasource "ds_ri_normalization" do
-  run_script $js_ri_cleanup, $ds_ri_recommendations, $param_currency_adjustment
+  run_script $js_ri_cleanup, $ds_ri_recommendations, $param_currency_adjustment, $ds_account_name
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_get_account_name", type: "javascript" do
+  parameters "account_id", "billing_centers", "org", "optima_host"
+  result "request"
+  code <<-EOS
+    // returns date formatted as string: YYYY-mm-dd
+    function getFormattedDailyDate(date) {
+      var year = date.getFullYear();
+      var month = (1 + date.getMonth()).toString();
+      month = month.length > 1 ? month : '0' + month;
+      var day = date.getDate().toString();
+      day = day.length > 1 ? day : '0' + day;
+      return year + '-' + month + '-' + day;
+    }
+    var start_date = getFormattedDailyDate(new Date(new Date().setDate(new Date().getDate() - 3)));
+    var end_date = getFormattedDailyDate(new Date(new Date().setDate(new Date().getDate() - 2)));
+    var request = {
+      auth: "auth_rs",
+      host: optima_host,
+      verb: "POST",
+      path: "/bill-analysis/orgs/" + org + "/costs/select",
+      body_fields: {
+        "dimensions": ["vendor_account_name"],
+        "granularity": "day",
+        "start_at": start_date,
+        "end_at": end_date,
+        "metrics": ["cost_nonamortized_unblended_adj"],
+        "billing_center_ids": _.compact(_.map(billing_centers, function(value){ return value.id})),
+        "limit": 1,
+        "filter": {
+          "dimension": "vendor_account",
+          "type": "equal",
+          "value": account_id[0]["account"]
+        }
+      },
+      headers: {
+        "User-Agent": "RS Policies",
+        "Api-Version": "1.0"
+      },
+      ignore_status: [400]
+    }
+  EOS
+end
+
+script "js_top_level_bc", type: "javascript" do
+  parameters "billing_centers"
+  result "filtered_billing_centers"
+  code <<-EOS
+    var filtered_billing_centers =_.reject(billing_centers, function(bc){ return bc.parent_id != null });
+  EOS
 end
 
 script "js_ri_cleanup", type: "javascript" do
-  parameters "ri_recos", "currency_adjustment"
+  parameters "ri_recos", "currency_adjustment", "ds_account_name"
   result "result"
   code <<-EOS
     var result = [];
+    accountName = ""
+    if (ds_account_name[0] != null) {
+      accountName = ds_account_name[0]['vendor_account_name']
+    }
     _.each(ri_recos, function(recos){
       var reco_details = recos["recommendationDetails"];
-
       _.each(reco_details, function(reco){
         result.push({
-          accountId: reco["accountId"],
+          AccountId: reco["accountID"],
```

Comment:
The key `AccountId` here should actually be `accountID` I believe? @jim-slattery-rs 

This key is referenced as a field on line 475